firedrake/slate/slac/compiler.py

```diff
@@ -114,14 +117,14 @@ def compile_expression(slate_expr, tsfc_parameters=None, coffee=False):
     # simply reuse the produced kernel.
     cache = slate_expr._metakernel_cache
     params = parameters["form_compiler"].copy()
-    if tsfc_parameters is not None:
+    params.update(parameters["slate_compiler"].copy())
+    if tsfc_parameters is not None:  # tsfc_parameters actually contains all compiler parameters
         params.update(tsfc_parameters)
```

Comment:
I think this means we have to ensure that the slate parameter names and the tsfc parameters don't overlap. Should we instead keep them separate as `{"slate": ..., "tsfc:" ...}` ?